keras_cv/layers/object_detection/iou_similarity.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+from keras_cv import bounding_box
+
+
+class IouSimilarity(tf.keras.layers.Layer):
+    """Computes a lookup table matrix containing the ious for a given pair of boxes.
+
+    The lookup vector is to be indexed by [`boxes1_index`,`boxes2_index`] if boxes
+    are unbatched and by [`batch`, `boxes1_index`,`boxes2_index`] if the boxes are
+    batched.
+
+    Users can configure to mask the result. `box1_mask` will mask `boxes1` and `box2_mask`
+    will mask `boxes2`, the result will be masked by `mask_val`. Boxes are masked when they're
+    out of the boundary, i.e., all their coodinates are less than 0. Users can pad ground truth
+    boxes with -1. in order for it to be masked here.
+
+    Args:
+      boxes1: a list of bounding boxes in 'corners' format. Can be batched or unbatched.
+      boxes2: a list of bounding boxes in 'corners' format. This should match the rank and
+        shape of boxes1.
+      bounding_box_format: a case-insensitive string.
+        For detailed information on the supported format, see the
+        [KerasCV bounding box documentation](https://keras.io/api/keras_cv/bounding_box/formats/).
+
+    Returns:
+      iou_lookup_table: a vector containing the pairwise ious of boxes1 and
+        boxes2.
+    """
+
+    def __init__(
+        self,
+        box1_format,
+        box2_format,
+        box1_mask=False,
+        box2_mask=False,
+        mask_val=-1,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.box1_format = box1_format
+        self.box2_format = box2_format
+        self.mask_val = mask_val
+        self.box1_mask = box1_mask
+        self.box2_mask = box2_mask
+        self.built = True
+
+    def call(self, boxes1, boxes2, boxes1_mask=None, boxes2_mask=None):
```

Comment:
Whats the specific use case of accepting explicit masks into call() instead of masking with -1s?